File: desafio1.dart
int numberDivisibleBy3or5(int numberForAnalysis) {
  final List<int> numbersValid = [];
  for (int index = 0; index < numberForAnalysis; index++) {
    final validations = [
      index % 3 == 0,
      index % 5 == 0,
    ];
    if (validations.contains(true)) numbersValid.add(index);
  }
  return numbersValid.reduce((acc, value) => acc += value);
}

void main(List<String> args) {
  print(numberDivisibleBy3or5(10));
}

File: desafio1.js
const numberDivisibleBy3or5 = (numberByAnalise) => {
  const numbersValid = [];
  for (let index = 1; index < numberByAnalise; index+=1 ) {
    const validations = [
      index % 3 === 0,
      index % 5 === 0,
    ];

    if(validations.includes(true)) numbersValid.push(index);
  }

  return numbersValid.reduce((acc, value) => acc += value,0);
}

console.log(numberDivisibleBy3or5(11));